src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -180,7 +142,7 @@ public String getResourceURI(IResource resource) {
 		List<IResource> results = new ArrayList<IResource>(list.getItems().size());
 		for (IResource resource : list.getItems()) {
 			try{
-				results.add(createVersion(resource, namespace, resource.getApiVersion()));
+				results.add(create(resource, namespace));
 			}catch(OpenShiftException e){
 				if(e.getStatus() != null){
```

Comment:
maybe OpenShiftException#hasStatus() would add explicitness to the API (e.getStatus() == null is easy to use but user dev needs to think at status possibly being null and check it, while #hasStatus() would unveil this explicitly right in the API)